server/resource_server/emails/verification/code/random_code_generator.go

```diff
@@ -0,0 +1,18 @@
+package code
+
+import (
+	"globe-and-citizen/layer8/server/resource_server/constants"
+	"math/rand"
+	"strconv"
+	"strings"
+)
+
+type RandomCodeGenerator struct{}
+
+func (g *RandomCodeGenerator) GenerateCode(emailAddress string) string {
```

Comment:
why create a struct for this? why not just a function?

Reply:
Because there are going to be multiple implementations of code generation, and the caller code should be insensitive to a particular one